### PR TITLE
workflow: update CI  to reflect the current state

### PR DESCRIPTION
**Pull request by 0xc0170**
As we moved away from Travis, we should update our workflow as well. Plus some other fixes in CI page.

### CODE REVIEW

**Comment by 0xc0170** [comment]
@saheerb Please review

**Comment by 0xc0170** [comment]
and @jamesbeyond also

**Review by saheerb — APPROVED** [review]

File: docs/contributing/guidelines/ci.md [hunk 1]
```diff
@@ -45,21 +34,24 @@ How it works:
...
 - **jenkins-ci/greentea-test** - Runs [greentea tests](../debug-test/greentea-for-testing-applications.html).
-- **jenkins-ci/mbed2-build-ARM** - Builds Mbed OS 2 with the [ARM compiler](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler). Related commands:
    - `python tools/build_release.py -p <target> -t <toolchain>`.
-- **jenkins-ci/mbed2-build-GCC_ARM** - Builds Mbed OS 2 with GCC_ARM.
-- **jenkins-ci/mbed2-build-IAR** - Builds Mbed OS 2 with IAR.
 - **jenkins-ci/unittests** - Runs [unit tests](../debug-test/unit-testing.html).
-- **tools-test-linux** - Tests tools work on Linux.
-- **tools-test-mac** - Tests tools work on macOS.
-- **tools-test-windows** - Tests tools work on Windows.
+
```
Comment (0xc0170) [inline docs/contributing/guidelines/ci.md]: this spaces should be removed, fixing now

Comment (0xc0170) [reply docs/contributing/guidelines/ci.md]: fixed in a5420d7

**Review by jamesbeyond — APPROVED** [review]
LGTM

**Pull request merged**